sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQuerySourceBase.java

```diff
@@ -128,13 +133,23 @@ protected TableSchema getSchema(PipelineOptions options) throws Exception {
   protected abstract void cleanupTempResource(BigQueryOptions bqOptions) throws Exception;
 
   @Override
+  public BoundedReader<T> createReader(PipelineOptions options) throws IOException {
```

Comment:
Why do we need this and MergedBoundedReader? Sources are always split by the runner before reading, so you don't need to implement createReader() on a source that is not the result of a split() call.

Reply:
I'm not sure if we do need it.  Is there actually a guarantee somewhere that says BoundedSource will never have `createReader` called on it?  The API design is confusing here, and doing a find-references on `createReader` do show it being called.

Reply:
Yeah, it is not documented, but it is true in practice for all runners, and it is the reason why removing the json reading codepaths with no replacement is safe (see my PR).

Reply:
ok cool.  I'll rebase once that other PR lands.